Assets/MixedRealityToolkit/Inspectors/Setup/BuildDeployWindow.cs

```diff
@@ -0,0 +1,1436 @@
+
+            var curScriptingBackend = PlayerSettings.GetScriptingBackend(BuildTargetGroup.WSA);
+            string appxDirectory = curScriptingBackend == ScriptingImplementation.IL2CPP ? $"AppPackages\\{PlayerSettings.productName}" : $"{PlayerSettings.productName}\\AppPackages";
+
+            try
+            {
+                AppPackageDirectories.Clear();
+                string[] buildList = Directory.GetDirectories(BuildDeployPreferences.AbsoluteBuildDirectory, "*", SearchOption.AllDirectories);
+                foreach (string appBuild in buildList)
+                {
+                    if (appBuild.Contains(appxDirectory) && !appBuild.Contains($"{appxDirectory}\\"))
+                    {
+                        AppPackageDirectories.AddRange(Directory.GetDirectories(appBuild));
+                    }
+                }
+
+                IEnumerable<string> selectedDirectories =
+                    from string directory in AppPackageDirectories
+                    orderby Directory.GetLastWriteTime(directory) descending
+                    select Path.GetFullPath(directory);
+                Builds.AddRange(selectedDirectories);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                // unused
+            }
+
+            UpdatePackageName();
+        }
+
+        private static string CalcMostRecentBuild()
+        {
+            UpdateBuilds();
+            DateTime mostRecent = DateTime.MinValue;
+            string mostRecentBuild = string.Empty;
+
+            foreach (var fullBuildLocation in Builds)
+            {
+                DateTime directoryDate = Directory.GetLastWriteTime(fullBuildLocation);
+
+                if (directoryDate > mostRecent)
+                {
+                    mostRecentBuild = fullBuildLocation;
+                    mostRecent = directoryDate;
+                }
+            }
+
+            return mostRecentBuild;
+        }
+
+        private void UpdatePortalConnections()
+        {
+            targetIps = new string[portalConnections.Connections.Count];
+
+            for (int i = 0; i < targetIps.Length; i++)
+            {
+                var connection = portalConnections.Connections[i];
+                string machineName = connection.MachineName;
+                if (string.IsNullOrEmpty(machineName) && IsLocalConnection(connection))
+                {
+                    connection.MachineName = IsHoloLensConnectedUsb ? HOLOLENS_USB : LOCAL_MACHINE;
+                }
+
+                targetIps[i] = machineName + " - " + connection.IP;
+            }
+
+            UwpBuildDeployPreferences.DevicePortalConnections = JsonUtility.ToJson(portalConnections);
+            lastSessionConnectionInfoIndex = CurrentConnectionIndex;
+            Repaint();
+        }
+
+        private static bool IsLocalConnection(DeviceInfo connection)
+        {
+            return connection.IP.Equals(LOCAL_MACHINE, StringComparison.OrdinalIgnoreCase) ||
```

Comment:
I think this was previously `.Contains` to account for a potential port in the IP address field